### PR TITLE
Fix "pgp encrypt" tracking

### PR DESCRIPTION
**Pull request by zapu**
"pgp encrypt" used to ask for tracking confirmation but was not actually publishing tracking statements, so I've added code to do that.

Now the issue is with NewResolveThenIdentify2WithTrack or how it's ran by PGP engines. I needed to use `AlwaysBlock:   true, NeedProofSet:  true`. If `AlwaysBlock` was `false`, it would just exit before user has any chance to confirm or deny. If `NeedProofSet` was `false`, it would early-exit when it found a cached result, but the early exit would not set any return variables, so the callers (PGP engines) would see that the identity was not confirmed (`confirmResult.IdentityConfirmed == false`).

So the issue right now is that previous behavior would not ask for confirmation when recipient was already being tracked, even locally. I can imagine there might be people relying on non-blocking behavior of `pgp encrypt`. With this PR, it still does not ask when person is tracked remotely and proofs didn't change (`▶ INFO Your view is up-to-date`), but the local cache skip is gone because of `NeedProofSet`.

cc @patrickxb @maxtaco I'll appreciate any help! Thanks!

### CODE REVIEW

**Comment by maxtaco** [comment]
Seems like this breaks `TestPGPDecryptSignedOther`

File: go/client/identify_ui.go [hunk 1]
```diff
@@ -23,30 +23,6 @@ func NewIdentifyTrackUIProtocol(g *libkb.GlobalContext) rpc.Protocol {
...
-	if err != nil {
-		return false
-	}
-
-	return currentStatus.LoggedIn
-}
-
-// NewIdentifyMaybeTrackUIProtocol returns either IdentifyUiProtocol with
-// IdentifyTrackUI or IdentifyUI depending if user is logged in or not.
-func NewIdentifyMaybeTrackUIProtocol(g *libkb.GlobalContext) rpc.Protocol {
```
Comment (patrickxb) [inline go/client/identify_ui.go]: My memory of this isn't crystal clear, but I believe this was in place to allow `keybase pgp encrypt` to work logged out.  Does it still work logged out?

    keybase pgp encrypt user -m "hi" --no-self

works for me logged out in production currently.

File: go/engine/pgp_encrypt.go [hunk 2]
```diff
@@ -190,20 +190,42 @@ func (e *PGPEncrypt) loadSelfKey() (*libkb.PGPKeyBundle, error) {
 }
 
 func (e *PGPEncrypt) verifyUsers(ctx *Context, assertions []string, loggedIn bool) ([]string, error) {
+	// Get current user for tracking. Ignore errors, but look out for
+	// me == nil, e.g. when user is not logged in (pgp encrypt can be
+	// used without Keybase account).
+	me, _ := libkb.LoadMe(libkb.NewLoadUserArg(e.G()))
```
Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: why is error ignored?

File: go/engine/pgp_encrypt.go [hunk 3]
```diff
@@ -190,20 +190,42 @@ func (e *PGPEncrypt) loadSelfKey() (*libkb.PGPKeyBundle, error) {
...
+		}
+
+		if me != nil && confirmResult.RemoteConfirmed {
+			targ := &TrackTokenArg{
+				Token:   ieng.TrackToken(),
+				Me:      me,
+				Options: keybase1.TrackOptions{},
+			}
+			teng := NewTrackToken(targ, e.G())
+			RunEngine(teng, ctx)
```
Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: I don't think you need this.

identify2_with_uid.go inserts track tokens

**Comment by zapu** [comment]
Before it was fixed just enough so unlogged users were not asked for remote tracking; but then it turned out that even when you said "yes" to remote track as a logged user, it would still not post tracking statements. This PR is an attempt to fix it, but still needs some work.

File: go/engine/pgp_encrypt.go (hunk 3 above)
Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: (and the track operation should be complete at this point...)

Comment (zapu) [inline go/engine/pgp_encrypt.go]: I might need to take another look then, last time I was working on it we thought that posting track/follow statement is just not there at all during `pgp encrypt`.

**Comment by patrickxb** [comment]
@zapu ok, just want to make sure that still works.  I guess there's no test for it, so we won't know if it's broken.

**Comment by zapu** [comment]
It looks like before this commit:

https://github.com/keybase/client/commit/f5c6ea3efd408412c12f7c825d113216a6604ea5

it would run `TrackEngine` to track recipients. It was removed in favor of using `NewResolveThenIdentify`. Was it sufficient at the time to just run that to post tracking statement?

The only places I could find that uses "follow" endpoint are track_token and unfollow engines. So I'd assume you have to call `TrackToken` to actually post tracking statements.

**Comment by zapu** [comment]
@patrickxb can you take a look again? thank you

File: go/engine/pgp_encrypt.go (hunk 3 above)
Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: Ok, I guess my memory failed me here and we do need to do what you added here.

File: go/engine/pgp_encrypt.go [hunk 4]
```diff
@@ -190,20 +190,42 @@ func (e *PGPEncrypt) loadSelfKey() (*libkb.PGPKeyBundle, error) {
...
+		confirmResult := ieng.ConfirmResult()
+		if !confirmResult.IdentityConfirmed {
+			return nil, libkb.IdentifyFailedError{Assertion: userAssert, Reason: "Not confirmed by user."}
+		}
+
+		if me != nil && confirmResult.RemoteConfirmed {
+			targ := &TrackTokenArg{
+				Token:   ieng.TrackToken(),
+				Me:      me,
+				Options: keybase1.TrackOptions{},
```
Comment (patrickxb) [inline go/engine/pgp_encrypt.go]: should this be `topts`?

**Comment by patrickxb** [comment]
I'd check that this is still fixed:

https://keybase.atlassian.net/browse/CORE-2329

**Pull request closed without merging**